generic_tracers/generic_COBALT.F90

```diff
@@ -4144,38 +4152,43 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
 
 !
 !-------------------------------------------------------------------------------------------------
-! 5: Sedimentary/coastal fluxes/transformations
+! 5: Sediment, coastal and ice dynamics
 !-------------------------------------------------------------------------------------------------
 !
 
-    !
-    ! Iceberg and "Coastal" iron and other nutrient input.
-    !
+    ! Nutrient inputs associated with icebergs/frozen runoff.  This is currently entered as a surface flux.  The
+    ! parameters "jfe_iceberg_ratio", "jno3_iceberg_ratio" and "jpo4_iceberg_ratio" are the ratios of nutrient input
+    ! per kg of runoff.  For iron, values can be set within the broad ranges discussed in Laufkotter et al. (2018).
+    ! These inputs are currently entered at the ocean surface, but they have defined within a 3D array to allow
+    ! eventual consideration of depth-dependent inputs.
     do j = jsc, jec ; do i = isc, iec !{
+       ! CAS: Is this check relevant for MOM6?
        if (grid_kmt(i,j) .gt. 0) then !{
           cobalt%jfe_iceberg(i,j,1) = cobalt%jfe_iceberg_ratio*max(frunoff(i,j),0.0)/rho_dzt(i,j,1)
           cobalt%jno3_iceberg(i,j,1) = cobalt%jno3_iceberg_ratio*max(frunoff(i,j),0.0)/rho_dzt(i,j,1)
           cobalt%jpo4_iceberg(i,j,1) = cobalt%jpo4_iceberg_ratio*max(frunoff(i,j),0.0)/rho_dzt(i,j,1)
-          cobalt%jfe_coast(i,j,1) = cobalt%fe_coast * mask_coast(i,j) * grid_tmask(i,j,1) / &
-               sqrt(grid_dat(i,j))
        endif !}
     enddo; enddo  !} i,j
+    ! CAS: Is the necessary?
     do k = 2, nk ; do j = jsc, jec ; do i = isc, iec   !{
```

Comment:
I don't thinks so, since these are initialized with 0s [here](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/blob/d57dbe98a033352d316a5ccd52afa18e0ff62eb0/generic_tracers/generic_COBALT.F90#L6694:L6696)

Reply:
Hi Gaby.  Thank you for adding the figures and sorry it has taken me so long to look at them.  I too am a bit surprised to not see a clearer progression from the small to the largest flux, but I think this may be partly due to the prominent boundary condition noise.  I will be good to continue testing this in combination with Theresa's boundary nudging fixes.